Build cloud functions (#9)
meilisearch/firestore-meilisearch: CONTRIBUTING.md

 We do not enforce a naming convention for the PRs, but **please use something descriptive of your changes**, having in mind that the title of your PR will be automatically added to the next [release changelog](https://github.com/meilisearch/meilisearch-js/releases/).

## Development Workflow

### Requirements

To run this project, you will need:

- Node >= 12
- Npm >= v7
- A google account
- Latest version of `firebase-tools` the Firebase CLI:
``` bash
npm install -g firebase-tools
```

### Setup

Sign in with your Google Account:
``` bash
firebase login
```
Enable the extension developer commands:
``` bash
firebase --open-sesame extdev
```
Install dependencies:
``` bash
npm run install:functions
```
Build the project:
``` bash
npm run build
```
Launch MeiliSearch instance:
``` bash
docker pull getmeili/meilisearch:latest # Fetch the latest version of MeiliSearch image from Docker Hub
docker run -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --master-key=masterKey --no-analytics=true
```
Launch emulator with param:
``` bash
firebase ext:dev:emulators:start --test-params=test-params-example.env --project=name-of-the-project
```
Launch emulator with dataset:
``` bash

> firebase ext:dev:emulators:start --test-params=test-params-example.env--import=dataset --project=name-of-the-project

Reviewer: ```suggestion
firebase ext:dev:emulators:start --test-params=test-params-example.env --import=dataset --project=name-of-the-project
```

3. [Create a new Git branch](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-and-deleting-branches-within-your-repository).
4. Review the [Development Workflow](#development-workflow) section that describes the steps to maintain the repository.
5. Make the changes on your branch.
6. [Submit the branch as a PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) pointing to the `main` branch of the main meilisearch-js repository. A maintainer should comment and/or review your Pull Request within a few days. Although depending on the circumstances, it may take longer.<br>
 We do not enforce a naming convention for the PRs, but **please use something descriptive of your changes**, having in mind that the title of your PR will be automatically added to the next [release changelog](https://github.com/meilisearch/meilisearch-js/releases/).

## Development Workflow

### Requirements

To run this project, you will need:

- Node >= 12
- Npm >= v7
- A google account
- Latest version of `firebase-tools` the Firebase CLI:
``` bash
npm install -g firebase-tools
```

> ### Setup

Reviewer: ```suggestion
### Setup <!-- omit in toc -->
```

<!-- /MarkdownTOC -->

## Assumptions

1. **You're familiar with [GitHub](https://github.com) and the [Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests) (PR) workflow.**
2. **You've read the MeiliSearch [documentation](https://docs.meilisearch.com) and the [README](/README.md).**
3. **You know about the [MeiliSearch community](https://docs.meilisearch.com/learn/what_is_meilisearch/contact.html). Please use this for help.**

## How to Contribute

1. Make sure that the contribution you want to make is explained or detailed in a GitHub issue! Find an [existing issue](https://github.com/meilisearch/meilisearch-js/issues/) or [open a new one](https://github.com/meilisearch/meilisearch-js/issues/new).
2. Once done, [fork the meilisearch-js repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) in your own GitHub account. Ask a maintainer if you want your issue to be checked before making a PR.
3. [Create a new Git branch](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-and-deleting-branches-within-your-repository).

> 4. Review the [Development Workflow](#workflow) section that describes the steps to maintain the repository.

Reviewer 2: ```suggestion
4. Review the [Development Workflow](#development-workflow) section that describes the steps to maintain the repository.
```

5. Make the changes on your branch.
6. [Submit the branch as a PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) pointing to the `main` branch of the main meilisearch-js repository. A maintainer should comment and/or review your Pull Request within a few days. Although depending on the circumstances, it may take longer.<br>
 We do not enforce a naming convention for the PRs, but **please use something descriptive of your changes**, having in mind that the title of your PR will be automatically added to the next [release changelog](https://github.com/meilisearch/meilisearch-js/releases/).

## Development Workflow

### Requirements

To run this project, you will need:

- Node >= 12
- Npm
- A google account

> Install or update to the latest version of the Firebase CLI:

Reviewer: This could stay in requirements but should be added in the bullet point

```suggestion
- Latest version of `firebase-tools` the Firebase CLI: 
```

The code sample example below can stay!

``` bash
npm install -g firebase-tools
```
Sign in with your Google Account:
``` bash
firebase login
```
Enable the extension developer commands:
``` bash
firebase --open-sesame extdev
```
You may need to install typescript in global to run this project:
``` bash
npm install typescript -g

> ```

Author: I add it because when Morgane try it she had to add typescript too.
Maybe I didn't add all the needed packages?

Install dependencies:
``` bash
npm run install:functions

> ```

Reviewer: These Should be in `### Setup`

### Setup

Build the project:
``` bash
npm run build
```
Launch MeiliSearch instance:
``` bash
docker pull getmeili/meilisearch:latest # Fetch the latest version of MeiliSearch image from Docker Hub
docker run -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --master-key=masterKey --no-analytics=true
```
Launch emulator with param:
``` bash
firebase ext:dev:emulators:start --test-params=test-params-example.env

> ```

Reviewer: This fails with the following error: 
```
Error: No currently active project.
To run this command, you need to specify a project. You have two options:
- Run this command with --project <alias_or_project_id>.
- Set an active project by running firebase use --add, then rerun this command.
To list all the Firebase projects to which you have access, run firebase projects:list.
```

Author: My bad I change this command because I was hoping you didn't need the the `--project` option

Author: Can you try it with the new command ?
I think you may need to create a Firebase project so I had to add it in the step too

Reviewer: It worked out with this cmd: 

```bash
firebase ext:dev:emulators:start --test-params=test-params-example.env --project=name-of-the-project
```
🔥  Thanks
